### PR TITLE
Toolbar icon update + scan icon

### PR DESCRIPTION
**Pull request by adsamcik**
So this is my proposal (size 18) for the new toolbar icon. I made in white with transparency and it is filled when scan is active. I think taking away color would be nice because colors are distracting and the scanner really shouldn't attract attention unless there is something new, which it still will with the original badge. This way there won't be any issues with the low pixel count as there are with the refresh badge.

I agree the filling is not an obvious indicator, but I think it's good enough that if user needs it, they can easily figure out what it means.

Depends on the previous progress changes.

### CODE REVIEW

File: src/lib/background/background.js [hunk 1]
```diff
@@ -38,7 +46,11 @@ export class Background {
 
     this.scanQueue = new ScanQueue();
     this.scanQueue.bindScanComplete(this._handleScanComplete.bind(this));
+    this.scanQueue.bindQueueStateChange(
+      this._handleScanQueueStateChange.bind(this),
+    );
 
+    browser.browserAction.setIcon({path: defaultIcon});
```
Comment (sneakypete81) [inline src/lib/background/background.js]: Semantically this should be done in _refreshIcon - keep all of the icon handling in one function.

File: src/lib/background/background.js [hunk 2]
```diff
@@ -82,7 +102,6 @@ export class Background {
     const updateCount = this.pageStore.getPageList()
       .filter(isItemChanged).length;
 
-    browser.browserAction.setIcon({path: activeIcon});
```
Comment (sneakypete81) [inline src/lib/background/background.js]: Make this function `refreshIcon(icon)`, and pass in either scanIcon or defaultIcon.

File: src/lib/background/background.js [hunk 3]
```diff
@@ -155,4 +174,30 @@ export class Background {
...
+      sendResponse(message);
+    } else {
+      browser.runtime.sendMessage(message);
+    }
+
+    if (stateData.state === scanQueueStateEnum.ACTIVE) {
+      browser.browserAction.setIcon({path: scanIcon});
+    } else {
+      browser.browserAction.setIcon({path: defaultIcon});
+    }
```
Comment (sneakypete81) [inline src/lib/background/background.js]: These should be calls to _refreshIcon too.

**Comment by adsamcik** [comment]
I haven't really revised the code on this one because I waited on your feedback for different icon.

**Comment by sneakypete81** [comment]
What you have looks good to me. After this is merged I'll look through the documentation to see which resolutions are currently recommended.

**Comment by adsamcik** [comment]
But which ones did you look at, the icons that are actually used are mostly invisible on the web. Those that have the refresh icon are unusable. On 100% zoom the size is 16x16 on fullHD. There is no way to fit that refresh icon in there.

**Comment by sneakypete81** [comment]
Ok I see, I previously tested with a HiDPI display, so it was using the 48px icons (and scaling them down to 32px). I've tried it now on a lower resolution display.

Overall I still prefer the original blue icons with the "scanning" circle in the bottom right corner.

I don't think the icon style should change depending on the display DPI, and I'd like to keep the coloured icons for consistency with the other places the icons are used. Having what looks like a subtle dot in the bottom-right corner of the icon is ok with me, and I find a bigger change to the arrow fill colour too distracting, particularly since in previous Update Scanner versions the fill colour was used to indicate whether there were changes detected. If it confuses even 1% of people, that's too many :-).

Most users shouldn't generally care whether scanning is occurring or not. If they do want to know, they can either check for the dot, or open the popup to make sure.

**Comment by adsamcik** [comment]
Ok that might be a good idea, to just replace it with simple dot or a circle on lower resolutions.

**Comment by sneakypete81** [comment]
I realise the icon sizes are a bit out (should be 16px/32px/64px). That's a result of these being nearly 10 years old from when Firefox had a slightly larger toolbar!

**Comment by adsamcik** [comment]
ok since I don't have as much time as I had, I gave up for now on the idea of unifying and converting the toolbar icons to svg and just updated the original icons with scan icon. The smallest one now has just a dot instead of progress indicator.

**Comment by sneakypete81** [comment]
Sounds good, thanks.

Can you please rebase this onto master? I'm getting conflicts when I try to rebase it.

**Comment by adsamcik** [comment]
I don't really get how are you having these issues. Both github (web) and my local git keep telling me there is nothing to rebase.

**Comment by sneakypete81** [comment]
> Also why rebase?

For a small project like this it keeps the commit history much more readable. 

Perhaps you're seeing conflicts because you merged mutiple branches into your branch? In general, if master moves forward then it's a good idea to rebase your branch rather than merging. 

See the first section of https://git-scm.com/book/en/v2/Git-Branching-Rebasing for a good explanation. In particular:
> Often, you’ll do this to make sure your commits apply cleanly on a remote branch — perhaps in a project to which you’re trying to contribute but that you don’t maintain. In this case, you’d do your work in a branch and then rebase your work onto origin/master when you were ready to submit your patches to the main project. That way, the maintainer doesn’t have to do any integration work — just a fast-forward or a clean apply.

**Comment by adsamcik** [comment]
I don't understand how is it less readable than if you squash-merge it. You would have single commit on top of master branch with all the changes from this and no conflicts to solve like with rebase. Also last time I spent good hour trying to solve all the conflicts that came with the rebase and I wasn't really happy about something like that when in the end it seemed to have the same effect as squash-merge.

**Comment by sneakypete81** [comment]
Fair enough, a squash and merge is fine in this case, since there's no need for more than one commit.

Thanks!

**Pull request merged**